### PR TITLE
Mobility GHC927 Upgrade Final

### PR DESCRIPTION
**Pull request by arjunkathuria**
This PR includes all the required changes to make this project upgrade to and build with GHC 9.2.7 and related tooling.

Changes include but not limited to:-

0. Use updated upstream repos like "nammayatri/common" etc.
1. Use updated juspay dependencies (which were in-turn patched to be able to build with GHC 9.2.7) like sequelize, hedis etc.
2. Code changes and refactoring needed to build after breaking changes across major library version upgrades for eg. aeson 1.x -> aeson 2.x and others (servant etc.).
3. Updated Nix flake build setup.
4. Patched tests code to be able to build with GHC 9.2.7 updated libraries and packages.
5. Adds some basic QOL improvements like an envrc file to make it work with direnv, updated gitignore, new cabal.project file etc.

### CODE REVIEW

File: src/EulerHS/KVConnector/Utils.hs [hunk 1]
```diff
@@ -159,11 +162,11 @@ getPKeyAndValueList table = do
       keyValueList = sortBy (compare `on` fst) k
       rowObject = A.toJSON table
   case rowObject of
-    A.Object hm -> foldl' (\ acc x -> (go hm x) : acc) [] keyValueList
+    A.Object hm -> DL.foldl' (\acc x -> [go hm x] <> acc) [] keyValueList
```
Comment (aravindgopall) [inline src/EulerHS/KVConnector/Utils.hs]: why not `cons`?

File: src/EulerHS/SqlDB/Types.hs [hunk 2]
```diff
@@ -125,12 +127,12 @@ class BeamRunner beM where
 
 instance BeamRunner BS.SqliteM where
   getBeamDebugRunner (NativeSQLiteConn conn) beM =
-    \logger -> SQLite.runBeamSqliteDebug logger conn beM
+    \logger -> SQLite.runBeamSqliteDebug (logger) conn beM
```
Comment (aravindgopall) [inline src/EulerHS/SqlDB/Types.hs]: `()` not required.

File: src/EulerHS/KVConnector/Utils.hs (hunk 1 above)
Comment (arjunkathuria) [reply src/EulerHS/KVConnector/Utils.hs]: i'm sure i probably had a better reason at the time, but right now looking at this i can only think of better semantics. 

`<>` in general means combine these, and it's just clearer at a glance (to my eyes).
`cons` also should do fine and probably be a bit more performant in this particular case if the function is run a lot.

Comment (aravindgopall) [reply src/EulerHS/KVConnector/Utils.hs]: This function does get invoked for every DB call IIRC. so can we move this to use `cons`

Comment (arjunkathuria) [reply src/EulerHS/KVConnector/Utils.hs]: updated.

File: src/EulerHS/SqlDB/Types.hs (hunk 2 above)
Comment (arjunkathuria) [reply src/EulerHS/SqlDB/Types.hs]: updated

**Pull request merged**